guides/index_lifecycle.md

```diff
@@ -0,0 +1,152 @@
+# Index Lifecycle
+This guide covers OpenSearch Java Client API actions for Index Lifecycle. You'll learn how to create, read, update, and delete indices in your OpenSearch cluster. We will also leverage index templates to create default settings and mappings for indices of certain patterns.
+
+## Setup
+
+In this guide, we will need an OpenSearch cluster with more than one node. Let's use the sample [docker-compose.yml](https://opensearch.org/samples/docker-compose.yml) to start a cluster with two nodes. The cluster's API will be available at `localhost:9200` with basic authentication enabled with default username and password of `admin:admin`.
+
+To start the cluster, run the following command:
+
+```bash
+cd /path/to/docker-compose.yml
+docker-compose up -d
+```
+
+## IndexData Class
+We will be using IndexData class which is a simple java class that stores data and methods
+
+```java
+  static class IndexData {
+    private String firstName;
+    private String lastName;
+
+    public IndexData() {
+    }
+
+    public IndexData(String firstName, String lastName) {
+      this.firstName = firstName;
+      this.lastName = lastName;
+    }
+
+    public String getFirstName() {
+      return firstName;
+    }
+
+    public void setFirstName(String firstName) {
+      this.firstName = firstName;
+    }
+
+    public String getLastName() {
+      return lastName;
+    }
+
+    public void setLastName(String lastName) {
+      this.lastName = lastName;
+    }
+
+    @Override
+    public String toString() {
+      return String.format("IndexData{first name='%s', last name='%s'}", firstName, lastName);
+    }
+  }
+
+```
+
+## Index Api Actions
+
+### Create a client using RestClientTransport
+
+Let's create a client instance to access this cluster:
+
+This code example uses basic credentials that come with the default OpenSearch configuration.If you’re using the Java client with your own OpenSearch cluster, be sure to change the code so that it uses your own credentials.
+
+```java
+final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
```

Comment:
@govegito this snippet it not valid for `main` branch since it should be using Apache HttpClient 5 APIs, I think the best option is to retarget the pull request to `2.x` branch and than have another pull request for main.

Reply:
Understood, retargeting it to 2.x branch.